src/App.js

```diff
@@ -1,16 +1,86 @@
 import { CssBaseline } from '@mui/material';
 import { ThemeProvider } from '@mui/material/styles';
-import React from 'react';
+import React, { useEffect, useCallback } from 'react';
 import Div100vh from 'react-div-100vh';
 import './App.css';
 import { initAnalytics, logPageView } from './analytics';
 import Head from './components/Head/Head';
 import MapPage from './components/MapPage/MapPage';
 import theme from './theme';
+import { useDispatch } from 'react-redux';
+import { userDataSlice } from 'reducers/user';
+
+/**
+ * Handle checking geolocation on an interval.
+ *
+ * @param {NodeJS.Interval} interval Interval to clear on callback run.
+ * @param {(status: boolean) => void} setter Setter to run when the callbacks run.
+ */
+function geolocationInterval(interval, setter) {
```

Comment:
We use this function in one place, and it knows too much about its dependencies. I think that using that logic inline would make it more readable and easier to understand, rather than abstracting it from its original flow

Reply:
I think my initial reasoning was to reduce duplication.